src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Compiler/AddExpressionLanguageProvidersPass.php

```diff
@@ -36,10 +36,10 @@ public function process(ContainerBuilder $container)
         }
 
         // security
-        if ($container->has('security.access.expression_voter')) {
-            $definition = $container->findDefinition('security.access.expression_voter');
+        if ($container->has('security.expression_language')) {
+            $definition = $container->findDefinition('security.expression_language');
             foreach ($container->findTaggedServiceIds('security.expression_language_provider', true) as $id => $attributes) {
-                $definition->addMethodCall('addExpressionLanguageProvider', array(new Reference($id)));
+                $definition->addMethodCall('registerProvider', array(new Reference($id)));
```

Comment:
This change should be applied to the 3.4 branch IMO, so that 3.4 does not trigger a deprecation warning when using the 4.1 component (registering it during the instantiation of the right service is a bugfix anyway, even though the bug was unlikely to happen before this PR because the `security.expression_language` was not injected anywhere else than in `security.access.expression_voter` by default)

Reply:
Will create a separate PR for 3.4 then :+1:

Reply:
See https://github.com/symfony/symfony/pull/26802.

Reply:
@symfony/mergers We should merge #26802 first (and propagate it to master) and then rebase this branch before merging it, so that this diff disappears.

Reply:
#26802 has now been merged up to master.

Reply:
I squashed my commits and rebased. Diff is gone now :+1: